modules/videoio/cmake/plugin.cmake

```diff
@@ -0,0 +1,59 @@
+function(ocv_create_videoio_plugin default_name target target_desc videoio_src_file)
+
+  set(name ${default_name} CACHE STRING "")
+  set(destination "" CACHE PATH "")
```

Comment:
Lets avoid using of generic names in CMake cache: "name", "destination".
Use:
- `OPENCV_PLUGIN_NAME`
- `OPENCV_PLUGIN_DESTINATION` (or even `OPENCV_PLUGIN_${OPENCV_PLUGIN_NAME}_DESTINATION`)

What is about install destination?